### ISSUE TITLE
Test `03145_non_loaded_projection_backup` has failed

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#67430]
https://s3.amazonaws.com/clickhouse-test-reports/0/a023f2c970e64fa3ea1623e4a5aa48e377f374c1/stateless_tests__release_.html

**Comment by KochetovNicolai** [comment]
Here is msan failure. Possibly relevant.

https://s3.amazonaws.com/clickhouse-test-reports/0/1480275ade1919651877e087fee3fda19924bc9f/stress_test__azure__msan_.html

```
2024.07.29 23:18:55.918423 [ 41829 ] {f0fead51-0858-42cd-9c7b-20ec0aef8667} <Debug> executeQuery: (from [::1]:42226) (comment: 03145_non_loaded_projection_backup.sh) check table tp_1 settings check_query_single_value_result = 0; (stage: Co
mplete)

...

2024.07.29 23:18:55.982577 [ 9257 ] {f0fead51-0858-42cd-9c7b-20ec0aef8667} <Fatal> : Logical error: 'One file must be mapped to one object on blob storage by path store/41d/41d32a12-8c36-47b1-89e2-f4626ef470e2/0_5_5_0/txn_version.txt in MergeTree tables, have 2.'.
2024.07.29 23:18:56.213485 [ 9257 ] {f0fead51-0858-42cd-9c7b-20ec0aef8667} <Fatal> : Stack trace (when copying this message, always include the lines below):

0. ./contrib/llvm-project/libcxx/include/exception:141: Poco::Exception::Exception(String const&, int) @ 0x000000003f69c1a9
1. ./build_docker/./src/Common/Exception.cpp:111: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000001fd8fb13
2. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x00000000080dde7d
3. DB::Exception::Exception<String const&, unsigned long>(int, FormatStringHelperImpl<std::type_identity<String const&>::type, std::type_identity<unsigned long>::type>, String const&, unsigned long&&) @ 0x000000000b4293ad
4. ./build_docker/./src/Storages/MergeTree/DataPartStorageOnDiskFull.cpp:108: DB::DataPartStorageOnDiskFull::getRemotePath(String const&, bool) const @ 0x0000000038863129
5. ./contrib/llvm-project/libcxx/include/string:1499: DB::checkDataPart(std::shared_ptr<DB::IMergeTreeDataPart const>, bool, bool&, std::function<bool ()>, bool)::$_0::operator()() const @ 0x0000000039752ff7
6. ./build_docker/./src/Storages/MergeTree/checkDataPart.cpp:440: DB::checkDataPart(std::shared_ptr<DB::IMergeTreeDataPart const>, bool, bool&, std::function<bool ()>, bool) @ 0x000000003974a8b6
7. ./contrib/llvm-project/libcxx/include/__tree:1055: DB::StorageMergeTree::checkDataNext(std::shared_ptr<DB::IStorage::DataValidationTasksBase>&) @ 0x0000000039807797
8. ./build_docker/./src/Interpreters/InterpreterCheckQuery.cpp:0: DB::(anonymous namespace)::TableCheckWorkerProcessor::transform(DB::Chunk&) @ 0x00000000353d980f
9. ./contrib/llvm-project/libcxx/include/__utility/swap.h:35: DB::ISimpleTransform::transform(DB::Chunk&, DB::Chunk&) @ 0x000000002074d181
10. ./build_docker/./src/Processors/ISimpleTransform.cpp:99: DB::ISimpleTransform::work() @ 0x000000003a21228b
11. ./contrib/llvm-project/libcxx/include/list:588: DB::ExecutionThreadContext::executeTask() @ 0x000000003a2825ad
12. ./build_docker/./src/Processors/Executors/PipelineExecutor.cpp:273: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000003a2515c6
13. ./build_docker/./src/Processors/Executors/PipelineExecutor.cpp:410: DB::PipelineExecutor::executeImpl(unsigned long, bool) @ 0x000000003a24f683
14. ./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:274: DB::PipelineExecutor::execute(unsigned long, bool) @ 0x000000003a24f0e8
15. ./build_docker/./src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:0: void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true, true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000003a28f49d
16. ./base/base/../base/wide_integer_impl.h:817: ThreadPoolImpl<std::thread>::worker(std::__list_iterator<std::thread, void*>) @ 0x000000001ffe3d6f
17. ./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:302: void* std::__thread_proxy[abi:v15007]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000001fff18b1
18. ? @ 0x00007f7a25588ac3
19. ? @ 0x00007f7a2561a850

**Comment by CurtizJ** [comment]
I couldn't reproduce. Most likely fixed by #67491.
